rclpy/src/rclpy/_rclpy.c

```diff
@@ -62,20 +62,19 @@ rclpy_get_sigint_guard_condition(PyObject * Py_UNUSED(self), PyObject * Py_UNUSE
   rcl_guard_condition_t * sigint_gc =
     (rcl_guard_condition_t *)PyMem_Malloc(sizeof(rcl_guard_condition_t));
   *sigint_gc = rcl_get_zero_initialized_guard_condition();
-  PyObject * pygc = PyCapsule_New(sigint_gc, "rcl_guard_condition_t", NULL);
   rcl_guard_condition_options_t sigint_gc_options = rcl_guard_condition_get_default_options();
 
   rcl_ret_t ret = rcl_guard_condition_init(sigint_gc, sigint_gc_options);
   if (ret != RCL_RET_OK) {
     PyErr_Format(PyExc_RuntimeError,
       "Failed to create guard_condition: %s", rcl_get_error_string_safe());
-    Py_DECREF(pygc);
     rcl_reset_error();
+    PyMem_Free(sigint_gc);
     return NULL;
   }
   g_sigint_gc_handle = sigint_gc;
   PyObject * pylist = PyList_New(2);
```

Comment:
FWIW, the Python [documentation](https://docs.python.org/3.6/c-api/list.html#c.PyList_New) for `PyList_New` says:

```
Return a new list of length len on success, or NULL on failure.
```

Which means we should check the return value of `PyList_New` as well (throughout this file).  I'm not sure if we want to do that as part of this PR or a different one, though.

Reply:
I'd rather do this in a few smaller PRs. I double checked all the functions in `_rclpy.c`; these are being used and can fail but are not being checked.

`PyList_New`
`PyCapsule_New` FYI @dirk-thomas 
`PyLong_FromUnsignedLongLong`
`PyMem_Malloc`
`PyUnicode_FromString` (not in web documentation but [source code says otherwise](https://github.com/python/cpython/blob/4f469c096628af730b17798d0ebfd8925bfde836/Objects/unicodeobject.c#L2101))
`PyLong_FromSize_t`
`PyObject_GetAtterString`
`PyLong_FromLongLong`
`PyCapsule_GetName`
`PyList_Append`
`PyObject_CallObject`
`PyTuple_New`
`PyTuple_SetItem`
`PyList_SetItem`
`PyImport_ImportModule`
`PyObject_SetAttrString`

Reply:
Ah, good, you were a lot more comprehensive than I was.  It's fine by me to do it in separate PRs; I'd suggest a tracking bug since there are so many of them.  I'll leave it up to you, though.

Reply:
Tracking bug #139